### PR TITLE
DROID-1691 Multispaces | Tech | Resetting space name and icon

### PR DESCRIPTION
**Pull request by uburoiubu**

### CODE REVIEW

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -289,25 +289,30 @@ class HomeScreenViewModel(
...
-            ).map { result ->
-                val obj = result.firstOrNull()
-                obj?.spaceIcon(urlBuilder, spaceGradientProvider) ?: SpaceIconView.Placeholder
-            }.flowOn(appCoroutineDispatchers.io).collect {
-                icon.value = it
-            }
+            spaceManager
+                .observe()
+                .flatMapLatest { config ->
+                    storelessSubscriptionContainer.subscribe(
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:297]: What happens if the subscription returns an error?

Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:297]: thanks, i'll add a catch block

File: presentation/src/main/java/com/anytypeio/anytype/presentation/settings/MainSettingsViewModel.kt [hunk 2]
```diff
@@ -41,53 +40,52 @@ class MainSettingsViewModel(
...
-                    )
-                },
-                profile = profile?.let {
-                    WorkspaceAndAccount.ProfileData(
-                        name = profile.name ?: "",
-                        icon = profile.profileIcon(urlBuilder, spaceGradientProvider)
+    val workspaceAndAccount = spaceManager
+        .observe()
+        .flatMapLatest { config ->
+            storelessSubscriptionContainer.subscribe(
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/settings/MainSettingsViewModel.kt:53]: idem

Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/settings/MainSettingsViewModel.kt:54]: added catch block

**Review by konstantiniiv — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/settings/MainSettingsViewModel.kt [hunk 3]
```diff
@@ -41,53 +40,52 @@ class MainSettingsViewModel(
...
+                        Relations.NAME,
+                        Relations.ICON_EMOJI,
+                        Relations.ICON_IMAGE,
+                        Relations.ICON_OPTION
                     )
-                }
-            )
+                )
+            ).map { result ->
+                val workspace = result.find { it.id == config.workspace }
```
[the remark this answers is not in the document]
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/settings/MainSettingsViewModel.kt:68]: done

**Pull request merged**